_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -97,6 +103,38 @@ typedef std::map<CodecKey, Handlers> CodecId2Handlers;
 
 static const CodecId2Handlers codecId2Handlers =
 {
+#if defined(MFX_ENABLE_USER_ENCODE)
+    {
+        {
+            CodecKey::MFX_CODEC_DUMMY_FOR_PLUGIN,
+            // .fei =
+            false
+        },
+        {
+            // .primary =
+            {
+                // .ctor =
+                nullptr,
+                // .query =
+                [](mfxSession s, mfxVideoParam *in, mfxVideoParam *out)
+                {
+                    assert(s->m_plgEnc);
+                    return s->m_plgEnc->Query(s->m_pCORE.get(), in, out);
+                },
+                // .queryIOSurf =
+                [](mfxSession s, mfxVideoParam *par, mfxFrameAllocRequest *request)
+                {
+                    assert(s->m_plgEnc);
+                    return s->m_plgEnc->QueryIOSurf(s->m_pCORE.get(), par, request, 0);
+                }
+            },
+            // .fallback =
+            {
```

Comment:
Looks like compiler just wants a constructor... Can you, please, check whether declaring constructor in Funcs will fix the issue and avoid this `{}`:
```
struct Funcs {
  Funcs() = default;
}
```

Reply:
There is no difference between empty and default constructors in this regard. The compiler still warns us.

Reply:
hm. strange. The trick forks for std::function somehow...